### ISSUE TITLE
spam filter (js beta) should teach lookaheads and lookbehinds

### ISSUE THREAD

**Issue by hbar1st** [freeCodeCamp/freeCodeCamp#55604]
### Describe the Issue

The description of the spam filter project currently mentions: "...learn about capture groups, positive lookaheads, negative lookaheads ..." but the project does not in fact teach positive or negative lookaheads.

It would be great if it did though!
Not sure if the fix for this is to fix the description or to add the missing lessons.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/#learn-regular-expressions-by-building-a-spam-filter

### Your code

N/A

### Expected behavior

Either change the description of the spam filter project to match what is currently taught or update the lessons to include the missing lookahead and lookbehind concepts.

### System

N/A

**Comment by naomi-lgbt** [comment]
Let's go ahead and teach them, then. 😁 

We will likely need to add a few new steps to teach positive and negative lookaheads and lookbehinds.

I'm not entirely sure where the best place is for those new steps tho.

**Comment by naomi-lgbt** [comment]
> even just start from step 36 (add it right at the end)

Sure, this works!

> Can I also add that we haven't seen how to use capturing groups properly? (like the syntax that lets you capture the group and re-use the pattern later on in your regex)

That should be tracked in a separate issue to avoid scope creep. 💜

**Comment by hbar1st** [comment]
Could you confirm if the capturing group idea(s) are taught later on though as I haven't going thru all the js beta projects. (So far, up to the Number Sorter, I have not seen it.) If it isn't covered, I'll definitely open another issue.

**Comment by naomi-lgbt** [comment]
I don't believe the syntax for replacing capture groups in strings is taught.

**Comment by ShaunSHamilton** [comment]
Just to mention, ~25% of safari users still cannot use lookbehinds: https://caniuse.com/js-regexp-lookbehind

So, a note mentioning that to Campers in/before the lesson would probably be a good idea to prevent a flood of forum posts.

**Comment by hbar1st** [comment]
In case the two issues end up interacting, here’s the one about back references  #55648

**Comment by ojeytonwilliams** [comment]
> So, a note mentioning that to Campers in/before the lesson would probably be a good idea to prevent a flood of forum posts.

We've had some impressively catastrophic failures when using them in the past, so it's important to QA these changes on the affected browsers.

**Comment by ahmaxed** [comment]
We are currently working on the new full stack curriculum and we are not accepting large enhancements to the current certifications. As a result, I am closing this issue.
